crates/api_models/src/admin.rs

```diff
@@ -1920,13 +1920,25 @@ pub struct BusinessProfileCreate {
     /// Whether to use the billing details passed when creating the intent as payment method billing
     pub use_billing_as_payment_method_billing: Option<bool>,
 
-    /// A boolean value to indicate if customer shipping details needs to be collected from wallet connector (Eg. Apple Pay, Google Pay, etc.)
+    /// A boolean value to indicate if customer shipping details needs to be collected from wallet
+    /// connector only if it is required field for connector (Eg. Apple Pay, Google Pay etc)
     #[schema(default = false, example = false)]
-    pub collect_shipping_details_from_wallet_connector: Option<bool>,
+    pub collect_shipping_details_from_wallet_connector_if_required: Option<bool>,
```

Comment:
why do we have to change the name of this field? shouldn't the fact that this field is optional convey `if_required`?

Reply:
`if_required ` is convey that shipping details will be collected from wallets like (Apple Pay or Google Pay) only the chosen connector requires it.